### DISCUSSION TITLE
Applying Transforms to both Image and Label without SubjectDataset

### DISCUSSION THREAD

**Discussion by Jason-Fung** [TorchIO-project/torchio#921] [Q&A]
Hello,

I am wondering if I can perform the same RandomAffine transformation to both an image and its corresponding label without instantiating my data into SubjectDataset? I've been trying to scour through the source code and couldn't find a clue on how to do it.

**Comment by fepegar** [comment]
Hi, @Jason-Fung. Have you checked the tutorials? https://github.com/fepegar/torchio/tree/main/tutorials

**Reply by Jason-Fung** [reply]
Yes I checked out the tutorials, however the tutorials primarily uses SubjectDatasets in order for the transformations to be done on both raw image and segmentation masks. For my application, I want to compose the functions for augmentation and somehow simultaneously transform both my raw image and segmentation mask without instantiating my dataset as a SubjectDataset.

**Reply by romainVala** [reply]
`SubjectDatasets`, is  the convenient way to loop over the subject. I do not see what the problem with it, but at minima, you need to use a  torchio `Subject`  this is the way it works. to associate the two inputs

**Reply by fepegar** [reply]
As @romainVala said, the most convenient way is putting your images in a `Subject`. The same spatial transforms will be applied to all images:

```python
import torch
import torchio as tio

image_tensor = torch.as_tensor((0.8, 0, 0, 0.2)).reshape(1, 2, 2, 1)
seg_tensor = image_tensor > 0.5

image = tio.ScalarImage(tensor=image_tensor)
seg = tio.LabelMap(tensor=seg_tensor)

subject = tio.Subject(image=image, seg=seg)
subject.plot()
```

![Figure_1](https://user-images.githubusercontent.com/12688084/176384470-a24c5e66-268b-4d16-a704-103b7e34593e.png)

```python
flip = tio.Flip(0)
flip(subject).plot()
```

![Figure_2](https://user-images.githubusercontent.com/12688084/176384503-ad2bae49-6877-446e-af3f-e49c56b18406.png)

**Reply by Jason-Fung** [reply]
Ok, I understand what to do now. Thank you guys very much for the help!
